### PR TITLE
fix: reduce method complexity in smtp_api

### LINKED ISSUE #49: Fix "method_complexity" issue in lib/smtpapi.rb
Method `escape_unicode` has a Cognitive Complexity of 7 (exceeds 5 allowed). Consider refactoring.

https://codeclimate.com/github/sendgrid/smtpapi-ruby/lib/smtpapi.rb#issue_59f66996439acf0001000016

### PR DESCRIPTION
**Pull request by sony-mathew**
Closes #49 

# Fixes # 

### Checklist
- [x] I have made a material change to the repo (functionality, testing, spelling, grammar)
- [x] I have read the [Contribution Guide] and my PR follows them.
- [x] I updated my branch with the master branch.
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have added necessary documentation about the functionality in the appropriate .md file
- [x] I have added in line documentation to the code I modified

### Short description of what this PR does:
- modifying `to_array` method and `escape_unicode` methods by splitting it into smaller methods
- changing `alias_method` to `alias`

If you have questions, please send an email to [Sendgrid](mailto:dx@sendgrid.com), or file a Github Issue in this repository.

### CODE REVIEW

**Comment by SendGridDX** [comment]
[CLA assistant check](https://cla.sendgrid.com/sendgrid/smtpapi-ruby?pullRequest=62) <br/>All committers have signed the CLA.

**Comment by SendGridDX** [comment]
[CLA assistant check](https://cla.sendgrid.com/sendgrid/smtpapi-ruby?pullRequest=62) <br/>Thank you for your submission, we really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla.sendgrid.com/sendgrid/smtpapi-ruby?pullRequest=62) before we can accept your contribution.<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla.sendgrid.com/check/sendgrid/smtpapi-ruby?pullRequest=62) it.</sub>

**Comment by thinkingserious** [comment]
Thank you @sony-mathew!

File: lib/smtpapi.rb [hunk 1]
```diff
@@ -115,44 +115,54 @@ def set_ip_pool(pool_name)
...
+      if n == 1
+        ch = (ch - 0x10000) / 0x400 + 0xD800
+      elsif n == 2
+        ch = (ch - 0x10000) % 0x400 + 0xDC00
+      end
+      format('%04x', ch)
+    end
+
+    def valid_value?(val)
+      val.nil? || (!val.is_a?(Integer) && val.empty?)
```
Comment (childish-sambino) [inline lib/smtpapi.rb:162]: Looks more like `invalid_value`.

**Comment by thinkingserious** [comment]
Closing until PR feedback is addressed.

**Pull request closed without merging**